bin/create_fortran_test.py

```diff
@@ -155,25 +161,131 @@ def create_test(test_name, json_name):
     stub_file_name=os.path.join(os.path.dirname(test_name), exercise+'.f90')
     create_stub(exercise, stub_file_name)
 
-# nice to have
-# TODO
-def add_meta_and_doc_file(test_name, json_name):
+
+def add_meta_and_doc_file(test_file_name, json_name):
     """
     create 
     
         .meta/tests.toml 
         .meta/config.json
         .docs/instructions.md  (based on ../problem-specifications/exercises/[EXERCISE]/description.md)
 
-    for test_name
+    for test_file_name
 
     Not yet implemented.
     """
-    test_dir_name = os.path.dirname(test_name)
+    test_dir_name = os.path.dirname(test_file_name)
+    exercise_name = os.path.basename(test_dir_name).replace('-', '_')
     meta_dir = os.path.join( test_dir_name, '.meta')
+    test_toml = os.path.join( meta_dir, 'tests.toml')
+    write_tests_toml(json_name, test_toml)
+
     doc_dir = os.path.join( test_dir_name, '.docs')
+    desc_file = os.path.join( os.path.dirname(json_name),  'description.md')
+    instruction_file =  os.path.join(doc_dir, 'instructions.md')
+    write_instructions(desc_file, instruction_file)
+    
+    meta_yaml = os.path.join( os.path.dirname(json_name),  'metadata.yml')
+    local_config_json = os.path.join(meta_dir, 'config.json')
+    write_config_json(exercise_name, meta_yaml, local_config_json)
```

Comment:
The code could have a little better separation of concerns. This function is handling only filepaths, and the `write_etc` functions are doing both `read` and `write`

I suggest to either pull the `read` logic up to this level, or to create extra functions for the read logic that return either `str` or `list(str)`

Reply:
Tried to do this in commit 240739e. Not quite sure if it is much more readable.